Scripts/check-examples

```diff
@@ -67,6 +67,9 @@
 ##   check
```

Comment:
This script snuck in within a previous PR (#92) and I didn't have the chance to look it over much at the time.

Despite my comments on #35, I have no concerns about using something like this script to test embedded examples, and as I commented elsewhere I don't actually think you're using lit here anyway, just some lit-like syntax.

My only immediate concern is having to maintain a copy of this script in every repo that wants to test embedded examples. Why don't we move it under https://github.com/dafny-lang/dafny/tree/master/Scripts instead, marked as experimental? We can just have this CI git clone dafny so it can use that copy. 

(In fact you could make this work with its current location, but I'd like to be more obvious that consumers outside of the core dafny repo are referencing the script)

Reply:
The script is customized - slightly different than the one in dafny-lang/dafny. I'd like to get these tests working & merged and in a separate step refactor the script to be useful in multiple locations.

Created issue #3688 to address this.

Reply:
Ah that's fine then, will continue discussion on https://github.com/dafny-lang/dafny/issues/3688